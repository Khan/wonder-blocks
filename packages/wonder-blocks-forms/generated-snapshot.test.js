// This file is auto-generated by gen-snapshot-tests.js
// Do not edit this file.  To make changes to these snapshot tests
// edit packages/wonder-blocks-forms/docs.md and run `npm run gen-snapshot-tests`.
import React from "react";
import renderer from "react-test-renderer";
import Dropdown from "./components/dropdown.js";

describe("wonder-blocks-forms", () => {
    it("example 1", () => {
        const example = (
            <Dropdown
                items={[
                    {label: "item 1", value: 1},
                    {label: "item 2", value: 2},
                    {label: "item 3", value: 3},
                ]}
            />
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 2", () => {
        const example = (
            <Dropdown
                show={true}
                items={[
                    {label: "item 1", value: 1},
                    {label: "item 2", value: 2},
                    {label: "item 3", value: 3},
                ]}
            />
        );
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("example 3", () => {
        class Parent extends React.Component {
            constructor(props) {
                super(props);
                this.state = {
                    show: false,
                    selection: null,
                };
            }

            handleOpen() {
                this.setState({show: true});
            }

            handleClose() {
                console.log("handle close");
                this.setState({show: true});
            }

            handleChange(selection: any) {
                this.setState({selection});
                this.setState({show: false});
            }

            render() {
                console.log(`show = ${this.state.show}`);
                return (
                    <Dropdown
                        controlled={false}
                        show={this.state.show}
                        selection={this.state.selection}
                        onOpen={() => this.handleOpen()}
                        onClose={() => this.handleClose()}
                        onChange={(e, selection) =>
                            this.handleChange(selection)
                        }
                        items={[
                            {label: "item 1", value: 1},
                            {label: "item 2", value: 2},
                            {label: "item 3", value: 3},
                            {label: "item 4", value: 4},
                        ]}
                    />
                );
            }
        }

        const example = <Parent />;
        const tree = renderer.create(example).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
