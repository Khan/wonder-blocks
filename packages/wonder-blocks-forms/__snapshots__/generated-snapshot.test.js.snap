// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wonder-blocks-forms example 1 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "border": "solid 1px blue",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 20,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 4,
        "position": "relative",
        "width": 200,
        "zIndex": 0,
      }
    }
  >
    Click Me!
  </div>
</div>
`;

exports[`wonder-blocks-forms example 2 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "border": "solid 1px blue",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 20,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 4,
        "position": "relative",
        "width": 200,
        "zIndex": 0,
      }
    }
  >
    Click Me!
  </div>
  <div
    className=""
    style={
      Object {
        "alignItems": "stretch",
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 0,
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <div
      className=""
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": "blue",
            "color": "white",
          },
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "fontSize": 20,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      item 1
    </div>
    <div
      className=""
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": "blue",
            "color": "white",
          },
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "fontSize": 20,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      item 2
    </div>
    <div
      className=""
      onClick={[Function]}
      style={
        Object {
          ":hover": Object {
            "background": "blue",
            "color": "white",
          },
          "alignItems": "stretch",
          "borderStyle": "solid",
          "borderWidth": 0,
          "boxSizing": "border-box",
          "display": "flex",
          "flexDirection": "column",
          "fontSize": 20,
          "margin": 0,
          "minHeight": 0,
          "minWidth": 0,
          "padding": 0,
          "position": "relative",
          "zIndex": 0,
        }
      }
    >
      item 3
    </div>
  </div>
</div>
`;

exports[`wonder-blocks-forms example 3 1`] = `
<div
  className=""
  style={
    Object {
      "alignItems": "stretch",
      "borderStyle": "solid",
      "borderWidth": 0,
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "margin": 0,
      "minHeight": 0,
      "minWidth": 0,
      "padding": 0,
      "position": "relative",
      "zIndex": 0,
    }
  }
>
  <div
    className=""
    onClick={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "border": "solid 1px blue",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "boxSizing": "border-box",
        "display": "flex",
        "flexDirection": "column",
        "fontSize": 20,
        "margin": 0,
        "minHeight": 0,
        "minWidth": 0,
        "padding": 4,
        "position": "relative",
        "width": 200,
        "zIndex": 0,
      }
    }
  >
    Click Me!
  </div>
</div>
`;
